server/plugin.go

```diff
@@ -229,82 +224,67 @@ func (p *Plugin) handleClientID(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 	}
 
-	return
+	return nil, http.StatusOK
 }
 
-func (p *Plugin) handleProductType(w http.ResponseWriter, r *http.Request) {
+func (p *Plugin) handleProductType(w http.ResponseWriter, r *http.Request) (error, int) {
 
 	userID := r.Header.Get("Mattermost-User-Id")
 	if userID == "" {
-		http.Error(w, "Not authorized", http.StatusUnauthorized)
-		return
+		return errors.New("cannot fetch Product Type. Missing 'Mattermost-User-Id' header"), http.StatusUnauthorized
 	}
 
 	config := p.getConfiguration()
 	if config == nil {
-		http.Error(w, "Cannot fetch configuration", http.StatusUnauthorized)
-		return
+		return errors.New("cannot fetch Product Type. Fetched configuration is empty"), http.StatusUnauthorized
 	}
 
 	w.Header().Set("Content-Type", "application/json")
 
 	if err := json.NewEncoder(w).Encode(ProductTypeResponse{
 		ProductType: p.getConfiguration().ProductType,
 	}); err != nil {
-		fmt.Println(err.Error())
-		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return fmt.Errorf("cannot fetch Product Type. An error occured while encoding the product type response: %w",
+			err), http.StatusInternalServerError
 	}
 
-	return
+	return nil, http.StatusOK
 }
 
-func (p *Plugin) handleRegisterMeetingFromOnlineVersion(w http.ResponseWriter, r *http.Request) {
+func (p *Plugin) handleRegisterMeetingFromOnlineVersion(w http.ResponseWriter, r *http.Request) (error, int) {
 
 	userID := r.Header.Get("Mattermost-User-Id")
 	if userID == "" {
-		fmt.Println("Request doesn't have Mattermost-User-Id header")
-		http.Error(w, "Not authorized", http.StatusUnauthorized)
-		return
+		return errors.New("cannot register meeting. Missing 'Mattermost-User-Id' header"), http.StatusUnauthorized
 	}
 
 	user, appErr := p.API.GetUser(userID)
 	if appErr != nil {
-		fmt.Println(appErr.Error())
-		http.Error(w, appErr.Error(), appErr.StatusCode)
-		return
+		return errors.New(fmt.Sprintf("cannot register meeting. An error occured while fetching user: %+v", appErr)), appErr.StatusCode
```

Comment:
Nit: You can simplicity these lines by using [`github.com/pkg/errors.Wrapf`](https://pkg.go.dev/github.com/pkg/errors?tab=doc#Wrapf)
```suggestion
		return errors.Wrapf(appErr, "cannot register meeting. An error occured while fetching user"), appErr.StatusCode
```

Reply:
Done :1: